Test1/ConcurrentPriorityQueue/ConcurrentPriorityQueue.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace ConcurrentPriorityQueue;
+
+using System.Collections;
+
+/// <summary>Represents a thread-safe max priority queue.</summary>
+/// <typeparam name="TElement">Specifies the type of elements in the queue.</typeparam>
+/// <typeparam name="TPriority">Specifies the type of priority associated with enqueued elements.</typeparam>
+public class ConcurrentPriorityQueue<TElement, TPriority>
+{
+    private readonly object lockObject = new();
+
+    // By the task we need to dequeue element with highest priority, but by default PriorityQueue dequeues element with the lowest one
+    // We hack that by reversing internal comparator
+    private readonly PriorityQueue<TElement, TPriority> queue =
+        new(Comparer<TPriority>.Create((a, b) => -Comparer.Default.Compare(a, b)));
```

Comment:
```suggestion
    private readonly PriorityQueue<TElement, TPriority> queue =
        new(Comparer<TPriority>.Create((a, b) => -a.CompareTo(b)));
```
Как-то не по ООП получилось. Вот так, например, явно видно, что приоритет реализует интерфейс `IComparable`. И метод этого интерфейса мы используем для получения компаратора.

Reply:
-0.5